### DISCUSSION TITLE
How does the live site set default flags such that they are visible in the options text box?

### DISCUSSION THREAD

**Discussion by sztomi** [compiler-explorer/compiler-explorer#5021] [Q&A] [answered]
In my configuration, I'm setting flags via `compiler.ID.options`. These flags are applied during all compilations, but they don't show up in the text box of compiler options. The live site seems to have some way of setting that, but I have trouble finding how it does it:

<img width="517" alt="image" src="https://user-images.githubusercontent.com/240594/236171322-a3be632e-79b5-4ed2-9c76-99c4769b66ed.png">

(my intention is mainly to set a standard level, but make it explicit to users and provide an easy way to override it). Can someone point me towards the config that does this?

**Comment by partouf** [comment, marked as answer]
As discussed in Discord; this doesn't happen automatically, `.options` are not visible for the user.

It has been on the table in the past and people filed issues for it here https://github.com/compiler-explorer/compiler-explorer/issues/1966 and here https://github.com/compiler-explorer/compiler-explorer/issues/3100

There is an implementation that is in the works here https://github.com/compiler-explorer/compiler-explorer/pull/5001 - but it might be a little while until it can do what you want.

**Reply by sztomi** [reply]
yep, thanks!
